### ISSUE TITLE
Failure to create a unique profile prefix when profiles starting with the same word exist

### ISSUE THREAD

**Issue by wondie** [gltn/stdm#126]
I already have a profile called **Rural**. Not knowing the consequence, I created another profile called **Rural Agriculture**. Then I created an entity named household and removed it again before finishing the wizard. When I clicked finish, I got the error below. 

This is because, the older profile **Rural** also has household entity. So when I removed the household entity, it also sent a delete command to the database. In my view, for new profile, if a user create entity and remove them, the delete command shouldn't be sent as the database table doesn't exist yet.  

The two issues here are;
1. Generation same prefix for profiles starting with the same word.
2. Deletion of entities (maybe the same for lookups, values, columns) even though they have no database table. 

```
Preparing configuration, please wait...
Configuration update started...
Scanning for changes in Informal_Settlement profile...
Removing redundant foreign key constraints...
Creating check_social_tenure_relationship_document_type entity...
Creating social_tenure_relationship_supporting_document entity...
Creating check_tenure_type entity...
Creating supporting_document entity...
Creating admin_spatial_unit_set entity...
Creating social_tenure_relationship entity...
Creating check_person_document_type entity...
Creating check_spatial_unit_document_type entity...
Creating check_gender entity...
Creating check_marital_status entity...
Creating check_household_relation entity...
Creating check_ownership_type entity...
Creating check_related_structures entity...
Creating check_recognition_status entity...
Creating check_utilities entity...
Creating Person_supporting_document entity...
Creating Person entity...
Creating Spatial_unit_supporting_document entity...
Creating Spatial_unit entity...
Creating related_structures entity...
Creating utilities entity...
Scanning for changes in Rural profile...
Removing redundant foreign key constraints...
Creating check_social_tenure_relationship_document_type entity...
Creating social_tenure_relationship_supporting_document entity...
Creating check_tenure_type entity...
Creating supporting_document entity...
Creating admin_spatial_unit_set entity...
Creating social_tenure_relationship entity...
Creating check_gender entity...
Creating check_farmer_document_type entity...
Creating check_sec_document_type entity...
Creating check_marital_status entity...
Creating check_saving_options entity...
Creating Garden entity...
Creating Sec_supporting_document entity...
Creating Sec entity...
Creating Household entity...
Creating Farmer_supporting_document entity...
Creating Farmer entity...
Creating savings entity...
Scanning for changes in Customary profile...
Removing redundant foreign key constraints...
Creating check_social_tenure_relationship_document_type entity...
Creating social_tenure_relationship_supporting_document entity...
Creating check_tenure_type entity...
Creating supporting_document entity...
Creating admin_spatial_unit_set entity...
Creating social_tenure_relationship entity...
Creating check_household_document_type entity...
Creating check_settlement_executive_committee_document_type entity...
Creating check_member_profile entity...
Creating Household_supporting_document entity...
Creating Household entity...
Creating composition entity...
Creating Parcel entity...
Creating Settlement executive committee_supporting_document entity...
Creating Settlement executive committee entity...
Scanning for changes in Formal profile...
Removing redundant foreign key constraints...
Creating check_social_tenure_relationship_document_type entity...
Creating social_tenure_relationship_supporting_document entity...
Creating check_tenure_type entity...
Creating supporting_document entity...
Creating admin_spatial_unit_set entity...
Creating social_tenure_relationship entity...
Creating check_person_document_type entity...
Creating Person_supporting_document entity...
Creating Person entity...
Creating Parcel entity...
Scanning for changes in Informal profile...
Removing redundant foreign key constraints...
Creating check_social_tenure_relationship_document_type entity...
Creating social_tenure_relationship_supporting_document entity...
Creating check_tenure_type entity...
Creating supporting_document entity...
Creating admin_spatial_unit_set entity...
Creating social_tenure_relationship entity...
Creating check_household_document_type entity...
Creating check_person_document_type entity...
Creating check_gender entity...
Creating check_income_sources entity...
Creating Household_supporting_document entity...
Creating Household entity...
Creating Structure entity...
Creating Person_supporting_document entity...
Creating Person entity...
Creating income_sources entity...
Scanning for changes in turkana profile...
Removing redundant foreign key constraints...
Creating check_social_tenure_relationship_document_type entity...
Creating social_tenure_relationship_supporting_document entity...
Creating check_tenure_type entity...
Creating supporting_document entity...
Creating admin_spatial_unit_set entity...
Creating social_tenure_relationship entity...
Creating check_document_type entity...
Creating check_gender entity...
Creating check_spatial_unit_document_type entity...
Creating check_party_document_type entity...
Creating Party_supporting_document entity...
Creating Party entity...
Creating Spatial unit_supporting_document entity...
Creating Spatial unit entity...
Creating Payment entity...
Scanning for changes in test profile...
Removing redundant foreign key constraints...
Creating check_social_tenure_relationship_document_type entity...
Creating social_tenure_relationship_supporting_document entity...
Creating check_tenure_type entity...
Creating supporting_document entity...
Creating admin_spatial_unit_set entity...
Creating social_tenure_relationship entity...
Creating Party entity...
Creating Sp entity...
Scanning for changes in Rural_Agriculture profile...
Removing redundant foreign key constraints...
Deleting Household entity...
(InternalError) cannot drop table ru_household because other objects depend on it
DETAIL:  constraint ru_farmer_household_fkey on table ru_farmer depends on table ru_household
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
 '\nDROP TABLE ru_household' {}
Failed to update configuration. Check error logs.
```

**Comment by gkahiu** [comment]
Hi Wondie,

Each profile has a unique prefix, see logic [here](https://github.com/gltn/stdm/blob/ref_config_mod/data/configuration/profile.py#L93) hence in your case, the initial rural will have a _ru_ prefix and the new rural agriculture will have a _rur_ prefix - did you confirm in pgAdmin regarding the prefix used in the tables, please check. Similarly, the system checks whether a table exists before attempting to delete using the _checkfirst=True_ flag as illustrated [here](https://github.com/gltn/stdm/blob/ref_config_mod/data/configuration/entity_updaters.py#L94).

From what I see, it seems like the household entity in rural profile is the **party** in the social tenure relationship definition and might have some **data**, can you confirm these two conditions?

**Comment by wondie** [comment]
I checked the database. No database table with prefix rur existed. However, when I re-run the wizard for a new profile rural agriculture, with no household, a rur prefix is added. So issue 1 is not an issue and is not the cause of the problem. 

The rural profile had no data. Moreover, I never deleted the household entity.  I only deleted the household in rural agriculture profile. My guess is it is deleting the household table using ru_household than checking for rur_household. So maybe that might be where the issue lies. In my view, for new profiles with no corresponding database, a delete command is not relevant.
